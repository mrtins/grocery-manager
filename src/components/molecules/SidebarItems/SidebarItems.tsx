import MenuItem from 'components/atoms/MenuItem/MenuItem';
import React from 'react';
import { HiLogout } from 'react-icons/hi';
import { SidebarRoute } from 'types';

interface SidebarItemsProps {
  itemsList: SidebarRoute[];
}

const SidebarItems = ({ itemsList }: SidebarItemsProps) => (
  <ul>
    {itemsList.map(({ title, icon, routeUrl }, i) => (
      <MenuItem
        data-testid="menu-item"
        key={i}
        title={title}
        icon={icon}
        routeUrl={routeUrl}
      />
    ))}

    <div className="bottom-6 absolute">
      <MenuItem title="Log out" icon={<HiLogout />} routeUrl="/logout" />
    </div>
  </ul>
);

export default SidebarItems;
