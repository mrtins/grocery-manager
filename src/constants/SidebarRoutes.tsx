import {
  HiAdjustments,
  HiCash,
  HiChartBar,
  HiCollection,
  HiHome,
  HiShoppingCart,
} from 'react-icons/hi';
import { SidebarRoute } from 'types';

const SidebarRoutes: SidebarRoute[] = [
  {
    title: 'Dashboard',
    icon: <HiHome />,
    routeUrl: '/',
  },
  {
    title: 'Compras',
    icon: <HiCash />,
    routeUrl: '/shopping-list',
  },
  {
    title: 'Estatisticas',
    icon: <HiChartBar />,
    routeUrl: '/stats',
  },
  {
    title: 'Categorias',
    icon: <HiCollection />,
    routeUrl: '/categories',
  },
  {
    title: 'Mercados',
    icon: <HiShoppingCart />,
    routeUrl: '/stores',
  },
  {
    title: 'Configurações',
    icon: <HiAdjustments />,
    routeUrl: '/settings',
  },
];

export default SidebarRoutes;
