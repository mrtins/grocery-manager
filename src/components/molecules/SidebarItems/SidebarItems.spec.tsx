import { render, screen, within } from '@testing-library/react';
import SidebarRoutes from 'constants/SidebarRoutes';
import SidebarItems from './SidebarItems';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
    };
  },
}));

describe('SidebarItems component', () => {
  it('should render SidebarItems with given itemList prop', () => {
    render(<SidebarItems itemsList={SidebarRoutes} />);

    const list = screen.getByRole('list', {
      name: '',
    });

    const { getAllByRole } = within(list);

    const items = getAllByRole('listitem');

    expect(items.length).toBe(7);
  });

  it('SidebarItems should match snapshot', () => {
    render(<SidebarItems itemsList={SidebarRoutes} />);

    const list = screen.getByRole('list', {
      name: '',
    });

    expect(list).toMatchSnapshot();
  });
});
