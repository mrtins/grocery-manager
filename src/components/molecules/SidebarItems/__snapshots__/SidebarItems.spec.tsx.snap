// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarItems component SidebarItems should match snapshot 1`] = `
<ul>
  <li
    class="false my-4 p-1 text-lg flex items-center justify-start cursor-pointer hover:border-l-blue-600 hover:text-blue-600 hover:font-medium hover:border-l-4"
  >
    <div
      class="ml-4 flex items-center"
      href="/"
    >
      <div
        class="text-blue-600"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z"
          />
        </svg>
      </div>
      <p
        class="ml-6"
        data-testid="menu-item-title"
      >
        Dashboard
      </p>
    </div>
  </li>
  <li
    class="false my-4 p-1 text-lg flex items-center justify-start cursor-pointer hover:border-l-blue-600 hover:text-blue-600 hover:font-medium hover:border-l-4"
  >
    <div
      class="ml-4 flex items-center"
      href="/shopping-list"
    >
      <div
        class="text-blue-600"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M4 4a2 2 0 00-2 2v4a2 2 0 002 2V6h10a2 2 0 00-2-2H4zm2 6a2 2 0 012-2h8a2 2 0 012 2v4a2 2 0 01-2 2H8a2 2 0 01-2-2v-4zm6 4a2 2 0 100-4 2 2 0 000 4z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <p
        class="ml-6"
        data-testid="menu-item-title"
      >
        Compras
      </p>
    </div>
  </li>
  <li
    class="false my-4 p-1 text-lg flex items-center justify-start cursor-pointer hover:border-l-blue-600 hover:text-blue-600 hover:font-medium hover:border-l-4"
  >
    <div
      class="ml-4 flex items-center"
      href="/stats"
    >
      <div
        class="text-blue-600"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 11a1 1 0 011-1h2a1 1 0 011 1v5a1 1 0 01-1 1H3a1 1 0 01-1-1v-5zM8 7a1 1 0 011-1h2a1 1 0 011 1v9a1 1 0 01-1 1H9a1 1 0 01-1-1V7zM14 4a1 1 0 011-1h2a1 1 0 011 1v12a1 1 0 01-1 1h-2a1 1 0 01-1-1V4z"
          />
        </svg>
      </div>
      <p
        class="ml-6"
        data-testid="menu-item-title"
      >
        Estatisticas
      </p>
    </div>
  </li>
  <li
    class="false my-4 p-1 text-lg flex items-center justify-start cursor-pointer hover:border-l-blue-600 hover:text-blue-600 hover:font-medium hover:border-l-4"
  >
    <div
      class="ml-4 flex items-center"
      href="/categories"
    >
      <div
        class="text-blue-600"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 3a1 1 0 000 2h6a1 1 0 100-2H7zM4 7a1 1 0 011-1h10a1 1 0 110 2H5a1 1 0 01-1-1zM2 11a2 2 0 012-2h12a2 2 0 012 2v4a2 2 0 01-2 2H4a2 2 0 01-2-2v-4z"
          />
        </svg>
      </div>
      <p
        class="ml-6"
        data-testid="menu-item-title"
      >
        Categorias
      </p>
    </div>
  </li>
  <li
    class="false my-4 p-1 text-lg flex items-center justify-start cursor-pointer hover:border-l-blue-600 hover:text-blue-600 hover:font-medium hover:border-l-4"
  >
    <div
      class="ml-4 flex items-center"
      href="/stores"
    >
      <div
        class="text-blue-600"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 1a1 1 0 000 2h1.22l.305 1.222a.997.997 0 00.01.042l1.358 5.43-.893.892C3.74 11.846 4.632 14 6.414 14H15a1 1 0 000-2H6.414l1-1H14a1 1 0 00.894-.553l3-6A1 1 0 0017 3H6.28l-.31-1.243A1 1 0 005 1H3zM16 16.5a1.5 1.5 0 11-3 0 1.5 1.5 0 013 0zM6.5 18a1.5 1.5 0 100-3 1.5 1.5 0 000 3z"
          />
        </svg>
      </div>
      <p
        class="ml-6"
        data-testid="menu-item-title"
      >
        Mercados
      </p>
    </div>
  </li>
  <li
    class="false my-4 p-1 text-lg flex items-center justify-start cursor-pointer hover:border-l-blue-600 hover:text-blue-600 hover:font-medium hover:border-l-4"
  >
    <div
      class="ml-4 flex items-center"
      href="/settings"
    >
      <div
        class="text-blue-600"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 4a1 1 0 00-2 0v7.268a2 2 0 000 3.464V16a1 1 0 102 0v-1.268a2 2 0 000-3.464V4zM11 4a1 1 0 10-2 0v1.268a2 2 0 000 3.464V16a1 1 0 102 0V8.732a2 2 0 000-3.464V4zM16 3a1 1 0 011 1v7.268a2 2 0 010 3.464V16a1 1 0 11-2 0v-1.268a2 2 0 010-3.464V4a1 1 0 011-1z"
          />
        </svg>
      </div>
      <p
        class="ml-6"
        data-testid="menu-item-title"
      >
        Configurações
      </p>
    </div>
  </li>
  <div
    class="bottom-6 absolute"
  >
    <li
      class="false my-4 p-1 text-lg flex items-center justify-start cursor-pointer hover:border-l-blue-600 hover:text-blue-600 hover:font-medium hover:border-l-4"
    >
      <div
        class="ml-4 flex items-center"
        href="/logout"
      >
        <div
          class="text-blue-600"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M3 3a1 1 0 00-1 1v12a1 1 0 102 0V4a1 1 0 00-1-1zm10.293 9.293a1 1 0 001.414 1.414l3-3a1 1 0 000-1.414l-3-3a1 1 0 10-1.414 1.414L14.586 9H7a1 1 0 100 2h7.586l-1.293 1.293z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <p
          class="ml-6"
          data-testid="menu-item-title"
        >
          Log out
        </p>
      </div>
    </li>
  </div>
</ul>
`;
